File: HealthKitDemo/Health/ViewController.m
//
//  ViewController.m
//  Health
//
//  Created by 叶增峰 on 16/8/10.
//  Copyright © 2016年 叶增峰. All rights reserved.
//

#import "ViewController.h"
#import "HealthManage.h"

@interface ViewController ()<UITableViewDelegate, UITableViewDataSource>

@property (nonatomic, strong) HKHealthStore *healthStore;

@property (nonatomic, strong) UITableView *tableView;

@property (nonatomic, strong) NSArray *dataSource;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.dataSource = @[@"get step count today", @"add 100 steps"];
    
    CGFloat screenWidth = UIScreen.mainScreen.bounds.size.width;
    CGFloat screenHeight = UIScreen.mainScreen.bounds.size.height;

    CGFloat y = UIApplication.sharedApplication.statusBarFrame.size.height + 44;
    
    self.tableView = [[UITableView alloc] initWithFrame:CGRectMake(0, y, screenWidth, screenHeight - y)];
    self.tableView.delegate = self;
    self.tableView.dataSource = self;
    self.tableView.tableFooterView = [UIView new];
    [self.view addSubview:self.tableView];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - HealthStepManage
- (void)getStepCountToday {
    [self getPermission];

    //获取当天步数
    [[HealthManage sharedInstance] getStepCountToday:^(int stepsResult){
        //主线程更新ui
        [[NSOperationQueue mainQueue] addOperationWithBlock:^{
            [self showAlertWithMessage:[NSString stringWithFormat:@"今日步数: %d", stepsResult]];
        }];
    }];
}

- (void)addSteps {
    [self getPermission];
    
    //写入步数
    NSDate *end = [NSDate date];
    NSDate *start = [end dateByAddingTimeInterval:(-60)];
    [[HealthManage sharedInstance] setStepCountWithSteps:100 startDate:start endDate:end completion:^(BOOL success) {
        [[NSOperationQueue mainQueue] addOperationWithBlock:^{
            [self showAlertWithMessage:[NSString stringWithFormat:@"添加步数%@", success ? @"成功" : @"失败"]];
        }];
    }];
}

- (void)getPermission {
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        [[HealthManage sharedInstance] getPermission];
    });
}

#pragma mark - UITableViewDelegate, UITableViewDataSource
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataSource.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];
    if(!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"cell"];
    }
    cell.textLabel.text = self.dataSource[indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    NSString *selected = self.dataSource[indexPath.row];
    if([selected isEqualToString:@"get step count today"]) {
        [self getStepCountToday];
    } else if([selected isEqualToString:@"add 100 steps"]) {
        [self addSteps];
    }
}

#pragma mark - AlertViewController
- (void)showAlertWithMessage:(NSString *)message {
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:nil message:message preferredStyle:UIAlertControllerStyleAlert];
    UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleCancel handler:nil];
    [alertController addAction:cancelAction];
    [self presentViewController:alertController animated:YES completion:nil];
}

@end

File: AVFoundationDemo/AVFoundationDemo/ViewController/首页/ViewController.swift
//
//  ViewController.swift
//  AVFoundationDemo
//
//  Created by YZF on 15/12/17.
//  Copyright © 2017年 Xiaoye. All rights reserved.
//

import UIKit
import SnapKit

class ViewController: UIViewController {
    
    enum Contents: String {
        case videoRecord_1 = "视频录制"
        case videoRecord_2 = "视频录制(多段视频并合并)"
        case videoRecord_3 = "视频录制(设置拍摄窗口大小)"
        case capturePhoto = "照片拍摄(对焦,前后置摄像头切换)"
        case wxVideoRecord = "微信拍摄"
        case musicPlayer_1 = "音乐播放器(控制中心，锁屏下可显示信息)"
        case musicPlayer_2 = "音乐播放器(边下边播，缓存)"
        case videoPlayer = "视频播放器(音量)"
    }
    
    @IBOutlet weak var tableView: UITableView! {
        didSet {
            tableView.delegate = self
            tableView.dataSource = self
            tableView.tableFooterView = UIView()
        }
    }
    
    let contents: [[Contents]] = [[.videoRecord_1, .videoRecord_2, .videoRecord_3, .capturePhoto, .wxVideoRecord],
                                  [.musicPlayer_1, .musicPlayer_2, .videoPlayer]]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
}

extension ViewController: UITableViewDelegate, UITableViewDataSource {
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return contents.count
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return contents[section].count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        var cell: UITableViewCell! = tableView.dequeueReusableCell(withIdentifier: "cell")
        if cell == nil {
            cell = UITableViewCell.init(style: .default, reuseIdentifier: "cell")
        }
        
        cell.textLabel?.text = contents[indexPath.section][indexPath.row].rawValue
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        var controller: UIViewController!
        switch contents[indexPath.section][indexPath.row] {
        case .videoRecord_1:
            controller = VideoRecordViewController_1()
        case .videoRecord_2:
            controller = VideoRecordViewController_2()
        case .videoRecord_3:
            controller = VideoRecordViewController_3()
        case .capturePhoto:
            controller = CapturePhotoViewController()
        case .wxVideoRecord:
            controller = WeiXinCameraViewController()
        case .musicPlayer_1:
            controller = MusicPlayerViewController()
            self.navigationController?.pushViewController(controller, animated: true)
            return
        case .musicPlayer_2:
            controller = MusicPlayerWithLoaderViewController()
            self.navigationController?.pushViewController(controller, animated: true)
            return
        case .videoPlayer:
            controller = VideoPlayerViewController()
            self.navigationController?.pushViewController(controller, animated: true)
            return
        }
        self.present(controller, animated: true, completion: nil)
    }
    
}

